akka-javasdk/src/main/scala/akka/javasdk/impl/http/JwtClaimsImpl.scala

```diff
@@ -198,16 +198,11 @@ class JwtClaimsImpl(jwtClaims: RuntimeJwtClaims) extends JwtClaims {
    * @return
    *   The boolean claim, if present. Returns empty if the claim is not a boolean or can't be parsed as a boolean.
    */
-  def getBoolean(name: String): Optional[lang.Boolean] = getString(name).flatMap((value: String) => {
-    def foo(value: String) = {
-      if (value.equalsIgnoreCase("true")) Optional.of(lang.Boolean.TRUE)
-      else if (value.equalsIgnoreCase("false")) Optional.of(lang.Boolean.FALSE)
-      Optional.empty
```

Comment:
I run into this while navigating the code. `foo` doesn't look great, but also the logic is maybe unnecessary complicated. Boolean.valueOf is already doing that for us.

There is a slight different though. If the property is not a valid boolean, it becomes a `None` in the previous impl. While in the new impl, it becomes `false`. Which I think it's more sensible since it's follow what Java does. 

Anyway, if that's not what we want I can change this PR and just rename `foo` to something more expressive.

Reply:
Hmm, now I see that the javadoc needs to be adapted. Or we should keep the old behavior.

Reply:
I think changing `foo` to something more expressive would be for sure better. 

Changing the behavior I'm not so sure. I think the signature reflects the intention, as you point out referring to the javadoc. All other methods follow the same idea. If is not parseable as expected we get a `None`. 

Coming back to the signature, I think it would be misleading/unexpected, for example, to parse a "trueu" as some' false.

Reply:
@franciscolopezsancho, I reverted the code, but remove the `foo` indirection.

Reply:
Roger that 👍 
Thanks for the call out @octonato